SPECS/protobuf/protobuf.spec

```
@@ -62,10 +64,11 @@ This contains protobuf python3 libraries.
 
 %build
 %{cmake} \
-    -Dprotobuf_BUILD_TESTS=OFF \
+    -Dprotobuf_BUILD_TESTS=ON \
     -Dprotobuf_ABSL_PROVIDER=package \
     -Dprotobuf_ABSL_MIN=20240116.0 \
     -Dprotobuf_BUILD_SHARED_LIBS=ON \
+    -Dprotobuf_USE_EXTERNAL_GTEST=ON \
```

Similar ask here - let's not build test for non-test builds.
```suggestion
    -Dprotobuf_ABSL_PROVIDER=package \
    -Dprotobuf_ABSL_MIN=20240116.0 \
    -Dprotobuf_BUILD_SHARED_LIBS=ON \
%if 0%{?with_check}
    -Dprotobuf_BUILD_TESTS=ON \
    -Dprotobuf_USE_EXTERNAL_GTEST=ON \
%else
   -Dprotobuf_BUILD_TESTS=OFF \
%endif
```